code/game/machinery/ecto_sniffer.dm

```diff
@@ -80,3 +81,39 @@
 	if(!ghost || isliving(ghost))
 		return
 	to_chat(ghost, "[FOLLOW_LINK(ghost, src)] <span class='nicegreen'>The coating of ectoplasmic residue you left on [src]'s sensors has decayed.</span>")
+
+
+/obj/item/ecto_alert
+	name = "ectoscopic alerter"
+	desc = "A small handheld device that listens for electroscopic transmissions."
+	icon = 'icons/obj/machines/research.dmi'
+	icon_state = "ecto_portable"
+	w_class = WEIGHT_CLASS_SMALL
+	var/on = TRUE
+	/// last time it recieved a signal
+	var/last_trigger = 0
+	/// how many minutes until the last trigger is cleared from the examinetext
+	var/clear_after = 5
+
+/obj/item/ecto_alert/Initialize(mapload)
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_GHOST_SUBMIT, .proc/activate)
+
+/obj/item/ecto_alert/attack_self(mob/user)
+	. = ..()
+	on = !on
+	to_chat(user, "<span class='notice'>You turn \the [src] [on ? "ON" : "OFF"].")
+
+/obj/item/ecto_alert/examine(mob/user)
+	. = ..()
+	if(last_trigger > world.time + clear_after MINUTES)  // only displays from the last 5 minutes
```

Comment:
should wrap (world.time + clear_after) with parenthesis and remove `MINUTES` as said above